examples/Example11/example11.cu

```diff
@@ -320,8 +321,20 @@ void example11(const vecgeom::cxx::VPlacedVolume *world, int numParticles, doubl
               << " number of hits: " << std::setw(4) << stats->scoring.hits;
     std::cout << std::endl;
 
+    // Check if only charged particles are left that are looping.
+    numElectrons = stats->inFlight[ParticleType::Electron];
+    numPositrons = stats->inFlight[ParticleType::Positron];
+    numGammas = stats->inFlight[ParticleType::Gamma];
+    if (numElectrons == previousElectrons && numPositrons == previousPositrons && numGammas == 0) {
```

Comment:
don't we have to check also BzFieldValue>0 here?

Reply:
We could, but we also need a cut-off mechanism in case there is no field. Actually, it might even be necessary to kill stuck photons (due to geometry errors)

Reply:
Exactly, but this type of check is typically done based on non-advancing, by setting a counter per track incremented whenever geometry returns a too small step.

Reply:
I'm against having a counter per track: It increases memory usage and checking per track, in every single step, is slower than one global check on the host (that already has the data)

Reply:
True, but how do we detect stuck tracks, you still have to do some accounting, no? And now we don't have a check for that. This does not concern this PR though.

Reply:
The same way this PR already works, even though we should probably allow more steps for gammas because they can go much further without interaction.